apps/download-service/src/app/modules/application/application.controller.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Body,
+  Controller,
+  Header,
+  Post,
+  Res,
+  Param,
+  UseGuards,
+} from '@nestjs/common'
+import { ApiOkResponse, ApiTags } from '@nestjs/swagger'
+import { Response } from 'express'
+import { ApplicationScope } from '@island.is/auth/scopes'
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import { GetApplicationPdfDto } from './dto/getApplicationPdf'
+import { ApplicationsApi } from '@island.is/clients/application'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApplicationScope.read)
+@ApiTags('application')
+@Controller('application')
+export class ApplicationController {
+  constructor(private readonly applicationApi: ApplicationsApi) {}
+
+  applicationApiWithAuth(auth: Auth, authorizationToken: string) {
+    auth.authorization = `Bearer ${authorizationToken}`
+    return this.applicationApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  @Post('/:applicationId')
+  @Header('Content-Type', 'application/pdf')
+  @ApiOkResponse({
+    content: { 'application/pdf': {} },
+    description: 'Get a PDF document for an application',
+  })
+  async getApplicationPdf(
+    @Param('applicationId') applicationId: string,
+    @CurrentUser() user: User,
+    @Body() resource: GetApplicationPdfDto,
+    @Res() res: Response,
+  ) {
+    const application = await this.applicationApiWithAuth(
+      user,
+      resource.__accessToken,
+    ).applicationControllerFindOne({
+      id: applicationId,
+    })
+
+    const externalData = application.externalData as {
+      sendApplication: {
+        data: {
+          pdfContent: string
+        }
+      }
+    }
+
+    const documentBase64 = externalData?.sendApplication?.data?.pdfContent
```

Comment:
Well I do agree it Is a bit clunky. Is there a straight forward way to add an attachment to the application in the template api ?

Reply:
Not yet, but I hear you're in the new Application System team. :)

Probably worth creating a service/helper to do it correctly (accept Buffer, create random key/filename, save file in S3 using PutObject, store full URL in application.attachments, return key so it can be stored in externalData/answer).